tests/phpunit/tests/rest-api/rest-attachments-controller.php

```diff
@@ -2262,4 +2262,45 @@ static function () {
 			}
 		);
 	}
+
+	/**
+	 * @ticket 55443
+	 */
+	public function test_image_sources_to_rest_response() {
+
+		$attachment_id = self::factory()->attachment->create_upload_object( $this->test_file );
+		$metadata      = wp_get_attachment_metadata( $attachment_id );
+		$request       = new WP_REST_Request();
+		$request['id'] = $attachment_id;
+		$controller    = new WP_REST_Attachments_Controller( 'attachment' );
+		$response      = $controller->get_item( $request );
+
+		$this->assertNotWPError( $response );
+
+		$data       = $response->get_data();
+		$mime_types = array(
+			'image/jpeg',
+		);
+
+		if ( wp_image_editor_supports( array( 'mime_type' => 'image/webp' ) ) ) {
+			array_push( $mime_types, 'image/webp' );
+		}
+
+		foreach ( $data['media_details']['sizes'] as $size_name => $properties ) {
+			if ( ! isset( $metadata['sizes'][ $size_name ]['sources'] ) ) {
+				continue;
+			}
+
+			$this->assertArrayHasKey( 'sources', $properties );
+			$this->assertIsArray( $properties['sources'] );
+
+			foreach ( $mime_types as $mime_type ) {
+				$this->assertArrayHasKey( $mime_type, $properties['sources'] );
+				$this->assertArrayHasKey( 'filesize', $properties['sources'][ $mime_type ] );
+				$this->assertArrayHasKey( 'file', $properties['sources'][ $mime_type ] );
+				$this->assertArrayHasKey( 'source_url', $properties['sources'][ $mime_type ] );
+				$this->assertNotFalse( filter_var( $properties['sources'][ $mime_type ]['source_url'], FILTER_VALIDATE_URL ) );
+			}
+		}
```

Comment:
Let's add one more assertion here that `$data['media_details']` does _not_ have the `sources` key, now that that is supposed to be unset in favor of having them as part of `$data['media_details']['sizes']['full']`.

Reply:
Thanks. This feedback is addressed.